src/www/javascript/opnsense_legacy.js

```diff
@@ -242,6 +242,10 @@ function hook_firewall_categories() {
             $(".rule").each(function(){
                 let is_selected = false;
                 $(this).data('category').split(',').forEach(function(item){
+                    if (selected_values.indexOf("(No Category)") > -1 && item === "") {
```

Comment:
doesn't take translations into account. when "no category" is always on top it's probably better to check if the first element of the selector is checked.

Reply:
Thanks for the heads up, silly mistake. See new commit.